src/webapp/components/table/BasicTable.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { Table, TableBody, TableCell, TableHead, TableRow, Link } from "@material-ui/core";
+import styled from "styled-components";
+import { Maybe } from "../../../utils/ts-utils";
+import { Selector } from "../selector/Selector";
+import i18n from "../../../utils/i18n";
+import { SelectorOption } from "../selector/utils/selectorHelper";
+
+interface BaseColumn {
+    value: string;
+    label: string;
+    underline?: boolean;
+}
+interface LinkColumn extends BaseColumn {
+    type: "link";
+}
+interface SelectorColumn extends BaseColumn {
+    type: "selector";
+    options: SelectorOption[];
+}
+
+export type TableColumn = BaseColumn | LinkColumn | SelectorColumn;
+interface BasicTableProps {
+    columns: TableColumn[];
+    rows: {
+        [key: TableColumn["value"]]: string;
+    }[];
+    onChange?: (cell: Maybe<string>, rowIndex: number, column: TableColumn["value"]) => void;
+    showRowIndex?: boolean;
+}
+
+export const BasicTable: React.FC<BasicTableProps> = React.memo(
+    ({ columns, rows, onChange = () => {}, showRowIndex = false }) => {
+        const Cell = (cell: string, rowIndex: number, column: TableColumn) => {
```

Comment:
Should we move Cell to another file and let it be its own component? Especially if we see its featrues growing in the future. Would it make sense to have different types of cells, like Link Cell, Selection Cell etc (and wrap the styles within that component) ?

Reply:
I agree that maybe it's better to have a Cell component instead of having it inside BasicTable

Reply:
I'll split them, is it ok to keep Cell component in the same folder?